### PR TITLE
CLDR-14797 Make most aliased items show as Missing in ST

### PR DESCRIPTION
**Pull request by macchiati**
CLDR-14797

- [ ] This PR completes the ticket.

See the discussion in the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Comment by macchiati** [comment]
Note: I'm not absolutely sure this is the right approach, so we shouldn't merge until after discussion.

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java
Comment (btangmu) [inline tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java:1269]: sourceLocale looks unused

Comment (macchiati) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java:1269]: It is unused . I just assigned it because that way it was easier to debug I think the compiler will just ignore it

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java
Comment (btangmu) [inline tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java:1715]: are there any callers of VettingViewer.getMissingStatus, other than this one, for which the adjustment would be wrong? if so, a brief explanatory comment might help, and in any case, maybe the adjustment should happen in getMissingStatus itself; it could depend on a boolean parameter if needed

there's a general problem that each component tends to have its own slight variation on an idea like "missing status", and the variations lead to inconsistencies that are sometimes unintended ... might there be an opportunity for more unification and encapsulation here?

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java
Comment (btangmu) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java:1269]: The compiler will issue a warning, depending on config -- one option would be if (DEBUG) ... println ... sourceLocale

Very minor quibble! I thought I'd better check anyway that the assignment wasn't for the sake of something intended but missing.

Comment (srl295) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java:1269]: quibble to the quibble: rather than if(DEBUG), use `logger.finest(() -> "sourceLocale:"+sourceLocale);`

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java
Comment (srl295) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java:1715]: there are multiple callers of VettingViewer.getMissingStatus, I think all within VettingViewer, but they all should give a consistent view of the missing status.

Perhaps getMissingStatus() shouldn't be public if it's only called within this object?

Comment (btangmu) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java:1715]: it's called by TestUtilities -- otherwise, only 3 callers, all in VettingViewer -- unless I'm leaving out a .jsp

**Comment by srl295** [comment]
Tests could still call it. Could be package level

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java
Comment (btangmu) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java:1715]: DataSection.java includes:
```
        CLDRFile.Status sourceLocaleStatus = new CLDRFile.Status();
        String sourceLocale = ourSrc.getSourceLocaleID(xpath, sourceLocaleStatus);
```
Should that change as well for consistency?

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java
Comment (macchiati) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java:1269]: I looked at https://www.geeksforgeeks.org/logger-finest-method-in-java-with-examples/, and that is a lot of function calls for something that is only needed when debugging. So I suggest we not entangle this PR, and that you put together a short doc about how and where we could deploy logging, and what the performance implications are, and we can discuss in the cldr infra meeting.

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java
Comment (macchiati) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java:1746]: 1. getMissingStatus is only used internally or in tests. I thought about pushing the special logic to getMissingStatus, but decided it was better to have the lower-level method be simpler.
2. I don't know the relation to DataSection.java; the call you have is for getSourceLocaleID. Do you mean "should it change to use the Extended version"? If that's what you mean, I don't think so.

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java
Comment (btangmu) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java:1292]: Steven's the expert on the new logging features but I think it would just entail adding one line in VettingViewer:
```
    private static final Logger logger = SurveyLog.forClass(VettingViewer.class);
```
Anyway that's low priority.

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java
Comment (btangmu) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java:1746]: >    ... the relation to DataSection.java ...

It's not clear to me. My concern is whether the status as indicated in the dashboard matches the status as shown in the data section and Info Panel. We can study that further after this is merged.

**Review by btangmu — APPROVED** [review]

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java
Comment (srl295) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java:1292]: yes, that will be documented in CLDR-5313

**Review by srl295 — APPROVED** [review]

**Comment by btangmu** [comment]
I'll have to remember this trick: ALLOW_MANY_COMMITS=true

**Comment by srl295** [comment]
requested CLA recheck wiht https://cla-assistant.io/check/unicode-org/cldr?pullRequest=1289

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java
Comment (macchiati) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/VettingViewer.java:1746]: It is used in testing; we could incorporate ICU's technique of marking methods with @internal or even come up with another annotation for methods that should only be public for testing purposes (@testing). But that would take more discussion in the cldr infra meeting.

I will add some better comments to the two methods.

**Pull request merged**
